### PR TITLE
Add model

### PR DESCRIPTION
**Pull request by brinxmat**
- Adds fileset, file and license

### CODE REVIEW

File: buildSrc/src/main/groovy/nvafilemodel.root.gradle [hunk 1]
```diff
@@ -0,0 +1,25 @@
+plugins{
```
Comment (brinxmat) [inline buildSrc/src/main/groovy/nvafilemodel.root.gradle:1]: If this file needs updating, please tell me

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -0,0 +1,48 @@
+[versions]
```
Comment (brinxmat) [inline gradle/libs.versions.toml:1]: This file is copied from datamodel-java and necessarily contains a lot of things that can be removed, however, before I do this, I think that it is interesting that nebula doesn't pick up on the fact that the definitions are unused. I guess that unused definitions are not included in gradle's dependency tree

File: nva-file-model/build.gradle [hunk 3]
```diff
@@ -0,0 +1,24 @@
+plugins{
+    id 'nvafilemodel.java-conventions'
+    id 'nvafilemodel.publish-maven'
+}
+
+dependencies {
+
+    compileOnly libs.nva.commons.core
```
Comment (brinxmat) [inline nva-file-model/build.gradle:8]: Here, on the other hand, I would have expected nebula-lint to complain

File: nva-file-model/src/main/java/no/unit/nva/file/model/File.java [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+import no.unit.nva.file.model.exception.MissingLicenseException;
+import nva.commons.core.JacocoGenerated;
+
+import java.time.Instant;
+import java.util.Objects;
+import java.util.UUID;
+
+import static java.util.Objects.isNull;
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
```
Comment (brinxmat) [inline nva-file-model/src/main/java/no/unit/nva/file/model/File.java:15]: This file requires a tidy, suggestions?

File: nva-file-model/src/main/java/no/unit/nva/file/model/FileSet.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
+public class FileSet {
+
+    public static final String FILES = "files";
+
+    @JsonProperty(FILES)
+    private final List<File> files;
+
+    @JsonCreator
+    public FileSet(@JsonProperty(FILES) List<File> files) {
```
Comment (brinxmat) [inline nva-file-model/src/main/java/no/unit/nva/file/model/FileSet.java:23]: I changed this file, it had a builder…

File: buildSrc/src/main/groovy/nvafilemodel.publish-maven.gradle [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+                        name = 'Orestis Gkorgkas'
+                        email = 'og@unit.no'
+                        organization = 'UNIT'
+                        organizationUrl = 'https://www.unit.no/'
+                    }
+                }
+                scm {
+                    connection = 'scm:git:git://github.com/BIBSYSDEV/nva-datamodel-java.git'
+                    developerConnection = 'scm:git:ssh://github.com/BIBSYSDEV/nva-datamodel-java.git'
+                    url = 'https://github.com/BIBSYSDEV/nva-datamodel-java/tree/main'
```
Comment (axthosarouris) [inline buildSrc/src/main/groovy/nvafilemodel.publish-maven.gradle]: update

File: buildSrc/src/main/groovy/nvafilemodel.publish-maven.gradle [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+                    fromResolutionOf('runtimeClasspath')
+                }
+                usage('java-runtime') {
+                    fromResolutionResult()
+                }
+            }
+            pom {
+                name = project.name
+                description = 'The java version of the NVA datamodel'
+                url = 'https://github.com/BIBSYSDEV/nva-datamodel-java'
```
Comment (axthosarouris) [inline buildSrc/src/main/groovy/nvafilemodel.publish-maven.gradle]: update

**Review by axthosarouris — CHANGES_REQUESTED** [review]

**Review by axthosarouris — APPROVED** [review]

**Pull request merged**